master/buildbot/reporters/bitbucketserver.py

```diff
@@ -105,44 +107,59 @@ def send(self, build):
                           code=response.code, content=content)
 
 
-class BitbucketServerPRCommentPush(http.HttpStatusPushBase):
+class BitbucketServerPRCommentPush(notifier.NotifierBase):
     name = "BitbucketServerPRCommentPush"
 
     @defer.inlineCallbacks
-    def reconfigService(self, base_url, user, password, text=None,
-                        verbose=False, **kwargs):
-        yield http.HttpStatusPushBase.reconfigService(
-            self, wantProperties=True, **kwargs)
-        self.text = text or Interpolate('Builder: %(prop:buildername)s '
-                                        'Status: %(prop:statustext)s')
+    def reconfigService(self, base_url, user, password, messageFormatter=None,
+                        verbose=False, debug=None, verify=None, **kwargs):
+        yield notifier.NotifierBase.reconfigService(
+            self, messageFormatter=messageFormatter, watchedWorkers=None,
+            messageFormatterMissingWorker=None, subject='', addLogs=False,
+            addPatch=False, **kwargs)
         self.verbose = verbose
         self._http = yield httpclientservice.HTTPClientService.getService(
             self.master, base_url, auth=(user, password),
-            debug=self.debug, verify=self.verify)
+            debug=debug, verify=verify)
 
-    @defer.inlineCallbacks
-    def send(self, build):
-        if build['complete'] and "pullrequesturl" in build['properties']:
-            yield self.sendPullRequestComment(build)
+    def checkConfig(self,base_url, user, password, messageFormatter=None,
+                    verbose=False, debug=None, verify=None, **kwargs):
+
+        notifier.NotifierBase.checkConfig(self,
+                                          messageFormatter=messageFormatter,
+                                          watchedWorkers=None,
+                                          messageFormatterMissingWorker=None,
+                                          subject='',
+                                          addLogs=False,
+                                          addPatch=False,
+                                          **kwargs)
+
+    def isMessageNeeded(self, build):
+        if 'pullrequesturl' not in build['properties']:
+            return False
+        return notifier.NotifierBase.isMessageNeeded(self, build)
+
+    def workerMissing(self, key, worker):
+        # nobody should call this, a comment is always associated to a change
+        pass
 
     @defer.inlineCallbacks
-    def sendPullRequestComment(self, build):
-        props = Properties.fromDict(build['properties'])
+    def sendMessage(self, body, subject=None, type=None, builderName=None,
+                    results=None, builds=None, users=None, patches=None,
+                    logs=None, worker=None):
+        props = Properties.fromDict(builds[0]['properties'])
```

Comment:
the event is always based on a build, so when buildComplete get called, it will have exactly one build and when buildsetComplete is called, if there is no builds (that passes the filter isMessageNeeded) then it will not be reported